packages/app/src/runner/SpecRunner.vue

```diff
@@ -186,6 +205,12 @@ onMounted(() => {
   eventManager.on('after:screenshot', () => {
     screenshotStore.setScreenshotting(false)
   })
+
+  eventManager.on('save:app:state', (state) => {
+    runnerUiStore.setIsSpecsListOpen(state.isSpecsListOpen)
```

Comment:
Hmm now I see this I wonder if we want a single `updateState` in the `runnerUiStore` that would more or less 1:1 map to `setPreferences.executeMutation`. Thoughts? Consider we have many more things to come like appX, appY, paneWidth etc - we likely want to have some pattern to consistently update the local Pinia store, as well at GraphQL.

Reply:
Eg, I wonder if we want to write some kind of abstraction like an `usePreferences` composables, that does something like:

```ts
function usePreferences (preference: 'autoScrollingEnabled' | ... others ..., value: any) {

  function update(preference: 'autoScrollingEnabled' | ... others ..., value: any) {
    runnerUiStore.setPreference(preference, value)
    setPreferences.executeMutation({ value: JSON.stringify({ [preference]: value ) })
  }

  return {
    update
  }
}
```

What do you think? Then we just do

```ts
const prefs = usePreferences()
prefs.update('autoScrollingEnabled', false)
```

And everything will stay in sync. We could strongly type this; the work has already been done in `packages/types/src/preferences.ts`.